dissect/target/plugins/os/unix/bsd/osx/network.py

```diff
@@ -0,0 +1,87 @@
+from __future__ import annotations
+
+import plistlib
+from typing import Iterator
+
+from dissect.target.helpers.record import MacInterfaceRecord
+from dissect.target.plugin import internal
+from dissect.target.plugins.general.network import NetworkPlugin
+
+
+class MacNetworkPlugin(NetworkPlugin):
+    SYSTEM = "/Library/Preferences/SystemConfiguration/preferences.plist"
+    DHCP = "/private/var/db/dhcpclient/leases"
+    plistlease = {}
+    plistnetwork = {}
```

Comment:
I think an `@cache` better shows your intent.

Reply:
Can you elaborate please?

Reply:
Use the `@cache` decorator instead of a manually maintained `dict`.